src/TLuaInterpreter.cpp

```diff
@@ -3029,51 +3029,101 @@ int TLuaInterpreter::setMainWindowSize(lua_State* L)
+    };
+
+    int s = 1;
+    if (lua_isstring(L, s) && !lua_isnumber(L, s)) {
+        windowName = QString::fromUtf8(lua_tostring(L, s));
+
+        if (!lua_isnumber(L, ++s)) {
+            lua_pushfstring(L, "setBackgroundColor: bad argument #%d type (red value 0-255 as number expected, got %s!)", s, luaL_typename(L, s));
+            return lua_error(L);
+        } else {
+            r = static_cast<int>(lua_tonumber(L, s));
+
+            if (!validRange(r)) {
+                lua_pushnil(L);
+                lua_pushfstring(L, "setBackgroundColor: bad argument #%d value (red value needs to be between 0-255, got %d!)", s, r);
+                return 2;
+            }
+        }
+    } else if (lua_isnumber(L, s)) {
+        r = static_cast<int>(lua_tonumber(L, s));
+
+        if (!validRange(r)) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "setBackgroundColor: bad argument #%d value (red value needs to be between 0-255, got %d!)", s, r);
+            return 2;
+        }
     } else {
-        x1 = lua_tonumber(L, 2);
+        lua_pushfstring(L, "setBackgroundColor: bad argument #%d type (window name as string, or red value 0-255 as number expected, got %s!)", s, luaL_typename(L, s));
+        return lua_error(L);
     }
-    double y1;
-    if (!lua_isnumber(L, 3)) {
-        lua_pushstring(L, "setBackgroundColor: wrong argument type");
-        lua_error(L);
-        return 1;
+
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "setBackgroundColor: bad argument #%d type (green value 0-255 as number expected, got %s!)", s, luaL_typename(L, s));
+        return lua_error(L);
     } else {
-        y1 = lua_tonumber(L, 3);
+        g = static_cast<int>(lua_tonumber(L, s));
+
+        if (!validRange(g)) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "setBackgroundColor: bad argument #%d value (green value needs to be between 0-255, got %d!)", s, g);
+            return 2;
+        }
     }
-    double x2;
-    if (!lua_isnumber(L, 4)) {
-        lua_pushstring(L, "setBackgroundColor: wrong argument type");
-        lua_error(L);
-        return 1;
+
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "setBackgroundColor: bad argument #%d type (blue value 0-255 as number expected, got %s!)", s, luaL_typename(L, s));
+        return lua_error(L);
     } else {
-        x2 = lua_tonumber(L, 4);
+        b = static_cast<int>(lua_tonumber(L, s));
+
+        if (!validRange(b)) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "setBackgroundColor: bad argument #%d value (blue value needs to be between 0-255, got %d!)", s, b);
+            return 2;
+        }
     }
-    double y2;
-    if (!lua_isnumber(L, 5)) {
-        lua_pushstring(L, "setBackgroundColor: wrong argument type");
-        lua_error(L);
-        return 1;
+
+    // if we get nothing for the alpha value, assume it is 255. If we get a non-number value, complain.
+    if (lua_gettop(L) <= 4) {
```

Comment:
Shouldn't this be `s` instead of `4` because if the window name is missing, alpha is the fourth argument but assumed as missing here.

Reply:
Well - if the window name is missing, meaning it's targetting the main window, alpha doesn't work anyway for them

Reply:
Seems to work still